net/tcp/tcp_conn.c

```diff
@@ -263,6 +260,20 @@ static FAR struct tcp_conn_s *
 static int tcp_selectport(uint8_t domain, FAR const union ip_addr_u *ipaddr,
                           uint16_t portno)
 {
+  static uint16_t g_last_tcp_port;
+
+  /* Generate port base dynamically */
+
+  if (g_last_tcp_port == 0)
+    {
+      g_last_tcp_port = clock_systime_ticks() % 32000;
+
+      if (g_last_tcp_port < 4096)
+        {
+          g_last_tcp_port += 4096;
+        }
+    }
+
```

Comment:
You still verify that this port number is unique, right?  With this change there is the real possibility that you could generate duplicate port numbers.  Logic must be included to avoid that.

Reply:
g_last_tcp_port will only be updated once if it is uninitialized

Reply:
Just initialize the port base instead of randomly generating the port every time...

Reply:
> 
> 
> You still verify that this port number is unique, right? With this change there is the real possibility that you could generate duplicate port numbers. Logic must be included to avoid that.

Has does that avoid duplicate port numbers?

Reply:
This can not be completely avoided if the time for creating the port between the two OS starts coincides exactly.